src/domain/opendatafabric/src/dtos/dtos_extra.rs

```diff
@@ -64,10 +64,13 @@ impl ExecuteTransformInput {
     /// from this input
     pub fn num_records(&self) -> usize {
         if let Some(new_offset) = self.new_offset {
-            if let Some(prev_offset) = self.prev_offset {
-                (new_offset - prev_offset) as usize
-            } else {
-                (new_offset + 1) as usize
+            #[cfg_attr(target_pointer_width = "64", allow(clippy::cast_possible_truncation))]
+            {
+                if let Some(prev_offset) = self.prev_offset {
+                    (new_offset - prev_offset) as usize
+                } else {
+                    (new_offset + 1) as usize
+                }
```

Comment:
I don't think there is any reason why `num_records` should return `usize`, and not `u64`.

Reply:
Agreed -- Done